sos/report/plugins/azure.py

```diff
@@ -66,4 +67,21 @@ def setup(self):
                 'openssl x509 -noout -text -in ' + crt_path
             ])
 
+
+class AzureEnhancedMonitoring(Azure, RedHatPlugin):
```

Comment:
This isn't a common way to add new functionality to sos - normally only one subclass of a plugin (with the appropriate tagging class) is executed per `sos report` run. Normally this kind of feature would either be merged into the existing (`Azure`) plugin, or if it's relatively orthogonal to the parent product then it may make sense for it to be a completely separate plugin in its own Python source file. I'm not 100% sure without testing what the plugin loader would do with this class - it's not necessarily wrong in Python terms, but it's not a model we promote or test for using the `sos` plugin class hierarchy.

There doesn't seem to be anything shared between the plugin and the parent in terms of state (just the trigger conditions in the class preamble) - if that was the case then that's a definite sign that the feature belongs in a parent class but my gut feeling is that this probably does belong as an addition to either `AzurePlugin` or if it's somehow specific to Red Hat on Azure then it ought to be in the `RedHatAzure` class.

Reply:
It's specific to Azure and it's AEM feature. I agree it might make sense to have a separate azure_aem plugin, 
I tested it and it worked :-)

What do you vote for? Separate plugin, given I can imagine to add more around AEM in future?

Reply:
Sure - but lots of plugins have optional parts that may or may not be present. This is already gated by the glob into `/var/lib` - if those paths aren't present then none of the extra collection runs and Azure users without AEM will not be troubled by it.

It's not so much what I vote for - the question is why does this need to be a separate plugin at all? Imho it should go into the parent `Azure` plugin class unless there is some reason not to do that (I can't see any from the information in the PR).